static/src/javascripts/projects/common/modules/user-prefs.spec.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import userPrefs from 'projects/common/modules/user-prefs';
+import { local } from 'lib/storage';
+
+class LocalStorageMock {
+    store: Object;
+
+    constructor() {
+        this.store = {};
+    }
+
+    clear() {
+        this.store = {};
+    }
+
+    getItem(key) {
+        return this.store[key];
+    }
+
+    setItem(key, value) {
+        this.store[key] = value.toString();
+    }
+
+    removeItem(key) {
+        this.store[key] = null;
+    }
+}
+
+const localStorage = new LocalStorageMock();
+
+describe('userPrefs - Client-side preferences', () => {
+    beforeAll(() => {
```

Comment:
It only runs each test spec (i.e. each `*.spec.js` file) in its own sandbox. Individual tests within a spec all run within the same sandbox, so we still need test-specific setup and teardown.